src/Profiles.ts

```diff
@@ -336,6 +347,224 @@ export class Profiles {
         return [updSession.ISession.user, updSession.ISession.password, updSession.ISession.base64EncodedAuth];
     }
 
+    public async getDeleteProfile() {
+        let zosmfProfile: IProfileLoaded;
+        const allProfiles: IProfileLoaded[] = this.allProfiles;
+        const profileNamesList = allProfiles.map((temprofile) => {
+            return temprofile.name;
+        });
+
+        if (!profileNamesList.length) {
+            vscode.window.showInformationMessage(localize("deleteProfile.noProfilesLoaded", "No profiles available"));
+            return;
+        }
+
+        const quickPickList: vscode.QuickPickOptions = {
+            placeHolder: localize("deleteProfile.quickPickOption", "Select the profile you want to delete"),
+            ignoreFocusOut: true,
+            canPickMany: false
+        };
+        const sesName = await vscode.window.showQuickPick(profileNamesList, quickPickList);
+
+        if (sesName === undefined) {
+            vscode.window.showInformationMessage(localize("deleteProfile.undefined.profilename",
+                "Operation Cancelled"));
+            return;
+        } else {
+            zosmfProfile = allProfiles.filter((temprofile) => temprofile.name === sesName)[0];
+        }
+
+        return zosmfProfile;
```

Comment:
This can be simplified again because of the `return` statement at the end of `if`

```suggestion

        return allProfiles.filter(temprofile => temprofile.name === sesName)[0];
```